src/engine/EventManager.cxx

```diff
@@ -16,6 +16,16 @@
 #include "microprofile.h"
 #endif
 
+template <typename TValue> bool contains(const std::vector<TValue> &c, const TValue &x)
+{
+  return std::find(std::begin(c), std::end(c), x) != std::end(c);
+}
+
+template <typename TValue, typename... Ts> bool contains(const std::map<Ts...> &c, const TValue &x)
+{
+  return c.find(x) != std::end(c);
+}
```

Comment:
The reason the standard doesn't include these functions is that most of the time, you will want to use the value if it exists afterwards. Instead, we should using `::find` everywhere we need a value

Reply:
what if we wanted only to check if it contains, which is really the case most of the time, and it's ugly.

Reply:
removed it